samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -732,89 +677,116 @@ static bool find_valid_bis_in_subgroup_bis(const struct bt_bap_base_subgroup *su
- * CONFIG_TARGET_BROADCAST_CHANNEL which makes it iterate over the next subgroup, and returns
- * `false` if this subgroup satisfies our CONFIG_TARGET_BROADCAST_CHANNEL.
+ * CONFIG_TARGET_BROADCAST_CHANNEL which makes it iterate over the next subgroup, and
+ * returns `false` if this subgroup satisfies our CONFIG_TARGET_BROADCAST_CHANNEL.
  */
-static bool find_valid_bis_in_subgroup_cb(const struct bt_bap_base_subgroup *subgroup,
-					  void *user_data)
+static bool subgroup_get_valid_bis_indexes_cb(const struct bt_bap_base_subgroup *subgroup,
+					      void *user_data)
 {
+	enum bt_audio_location subgroup_chan_allocation = BT_AUDIO_LOCATION_MONO_AUDIO;
+	enum bt_audio_location bis_chan_allocation = BT_AUDIO_LOCATION_MONO_AUDIO;
 	enum bt_audio_location chan_allocation;
+	bool subgroup_chan_allocation_available = false;
+	bool bis_chan_allocation_available = false;
 	struct bt_audio_codec_cfg codec_cfg;
-	uint32_t *bis_indexes = user_data;
+	uint32_t bis_indexes = 0;
 	int err;
 
-	/* We only want indexes from a single subgroup, so reset between each of them*/
-	*bis_indexes = 0U;
-
 	err = bt_bap_base_subgroup_codec_to_codec_cfg(subgroup, &codec_cfg);
 	if (err != 0) {
 		printk("Could not get codec configuration: %d\n", err);
 
-		return true;
+		return true; /* continue to next subgroup */
 	}
 
-	err = bt_audio_codec_cfg_get_chan_allocation(&codec_cfg, &chan_allocation, false);
-	if (err != 0) {
-		printk("Could not find subgroup channel allocation: %d - Looking in the BISes\n",
-		       err);
-
-		/* Find chan alloc in BIS */
-		if (find_valid_bis_in_subgroup_bis(subgroup, bis_indexes)) {
-			/* Found BISes with correct channel allocation */
-			return false;
-		}
+	/* Get channel allocation at subgroup level */
+	err = bt_audio_codec_cfg_get_chan_allocation(&codec_cfg, &subgroup_chan_allocation, false);
+	if (err == 0) {
+		printk("Channel allocation (subgroup level) 0x%x\n", subgroup_chan_allocation);
+		subgroup_chan_allocation_available = true;
+	} else if (err == -ENODATA) {
+		printk("No channel allocation data in subgroup\n");
 	} else {
-		/* If the subgroup contains a single channel, then we just grab the first BIS index
-		 */
-		if (bt_audio_get_chan_count(chan_allocation) == 1 &&
-		    chan_allocation == CONFIG_TARGET_BROADCAST_CHANNEL) {
-			uint32_t subgroup_bis_indexes;
+		/* subgroup error */
+		return true; /* continue to next subgroup */
+	}
 
-			/* Set bis_indexes to the first bit set */
-			err = bt_bap_base_subgroup_get_bis_indexes(subgroup, &subgroup_bis_indexes);
-			if (err != 0) {
-				/* Should never happen as that would indicate an invalid
-				 * subgroup If it does, we just parse the next subgroup
-				 */
-				return true;
-			}
+	/* Keep base code id (used in bt_audio_codec_cfg_get_chan_allocation at bis level) */
+	struct bis_channel_allocation_data data = {.cnt = 0,
+						   .bap_base_codec_id = {.id = codec_cfg.id,
+									 .cid = codec_cfg.cid,
+									 .vid = codec_cfg.vid}};
```

Comment:
```suggestion
	/* Keep base code id (used in bt_audio_codec_cfg_get_chan_allocation at bis level) */
	struct bis_channel_allocation_data data = {.cnt = 0,
						   .bap_base_codec_id = {.id = codec_cfg.id,
									 .cid = codec_cfg.cid,
									 .vid = codec_cfg.vid,},};
```

If you use clang-format, please end structs with a final `,` so that it formats it correctly instead of these weird formats :D

Reply:
Kind of moot, as this struct is pretty useless as described above :)